spark/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -389,7 +399,9 @@ public EncryptedOutputFile newOutputFile(PartitionKey key) {
     public abstract void write(InternalRow row) throws IOException;
 
     public void writeInternal(InternalRow row)  throws IOException {
-      if (currentRows % ROWS_DIVISOR == 0 && currentAppender.length() >= targetFileSize) {
+      //TODO: ORC file now not support target file size
```

Comment:
What exception you see if when running with ORC? I would have thought that it should work fine.
Also I think it is  disabled for Avro also now.

Reply:
Yes in Avro file is work,  cannot get the file length before closed the  ORC file, I have fix it.